tests/functional/test_freeze.py

```diff
@@ -1008,15 +1034,13 @@ def test_freeze_pep610_editable(script: PipTestEnvironment) -> None:
     is correctly frozeon as editable.
     """
     pkg_path = _create_test_package(script, name="testpkg")
-    result = script.pip("install", pkg_path)
+    result = script.pip("install", str(pkg_path))
     direct_url_path = get_created_direct_url_path(result, "testpkg")
     assert direct_url_path
     # patch direct_url.json to simulate an editable install
-    with open(direct_url_path) as f:
-        direct_url = DirectUrl.from_json(f.read())
+    direct_url = DirectUrl.from_json(direct_url_path.read_text())
```

Comment:
Unrelated to the `Path` change, but shouldn't we specify an encoding here? (And below).

Reply:
cc @sbidoul I’m not sure if platform-dependant encoding is intended here.

Reply:
It is was not intended, no. I'll do a PR to fix this (it should not matter much for the tests in practice but I spotted one place in the code where it could be an issue).